src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -515,6 +517,50 @@ impl AstDisplay for AlterObjectRenameStatement {
 }
 impl_display!(AlterObjectRenameStatement);
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum AlterIndexOptionsList {
+    Set(Vec<SqlOption>),
+    Reset(Vec<Ident>),
+}
+
+/// `ALTER INDEX ... {RESET, SET}`
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AlterIndexOptionsStatement {
+    pub index_name: ObjectName,
+    pub if_exists: bool,
+    pub options: AlterIndexOptionsList,
+}
+
+impl AstDisplay for AlterIndexOptionsStatement {
+    fn fmt(&self, f: &mut AstFormatter) {
+        f.write_str("ALTER INDEX ");
+        f.write_node(&self.index_name);
+        f.write_str(" ");
+        if self.if_exists {
+            f.write_str("IF EXISTS ");
+        }
```

Comment:
I think this puts the `IF EXISTS` clause in the wrong spot. Should go before the name, no?